dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java

```diff
@@ -331,10 +331,21 @@ else if (fd.getDataLen() == 8)
                 {
                     this.buffer.putLong(value);
                 }
+                else if (fd.getDataLen() < 8 && fd.getDataLen() > 0)
+                {
+                    long lastByteIdx = fd.getDataLen() -1;
```

Comment:
does this need to be a long?
Actually, I see it's used in the bit masking and shift in line 342